### PR TITLE
Auto compile address map

### PR DESCRIPTION
**Pull request by andywingo**
This PR implements the auto-compilation strategy discussed in #186.

### CODE REVIEW

File: src/apps/lwaftr/address_map.lua [hunk 1]
```diff
@@ -272,6 +285,77 @@ function compile(file)
    return attach_lookup_helper(builder:build())
 end
 
+local verbose = os.getenv('SNABB_LWAFTR_VERBOSE') or true
```
Comment (andywingo) [inline src/apps/lwaftr/address_map.lua]: this "or true" is just a temporary thing and probably shouldn't be there.  probably we should put a log function in the utils, unless snabb gives us something nice already

File: src/apps/lwaftr/address_map.lua [hunk 2]
```diff
@@ -272,6 +285,77 @@ function compile(file)
...
+      log('error saving compiled address map %s: %s', compiled, err)
+   end
+
+   -- Done.
+   return map
+end
+
+local function mktemp(name, mode)
+   if not mode then mode = "rusr, wusr, rgrp, roth" end
+   t = math.random(1e7)
```
Comment (dpino) [inline src/apps/lwaftr/address_map.lua]: Forgot to localize `t`.

File: src/apps/lwaftr/rangemap.lua [hunk 3]
```diff
@@ -144,23 +148,109 @@ function RangeMap:lookup(k)
...
+   -- OK!
+   map.entries = ffi.cast(ffi.typeof('$*', self.entry_type),
+                          ffi.cast('uint8_t*', mem) + offset)
+   map.size = size
+   map.binary_search = binary_search.gen(map.size, map.entry_type)
+   map.mtime_sec = header.mtime_sec
+   map.mtime_nsec = header.mtime_nsec
+   map = setmetatable(map, { __index = RangeMap })
+
+   ffi.gc(map.entries, function (ptr) S.munmap(mem, size) end)
```
Comment (dpino) [inline src/apps/lwaftr/rangemap.lua]: I think `ptr` wouldn't necessary here.

**Comment by dpino** [comment]
Looks great. LGTM!

**Comment by andywingo** [comment]
Tx for review :)

**Pull request merged**